core/src/main/java/nu/yona/server/util/Require.java

```diff
@@ -0,0 +1,63 @@
+/*******************************************************************************
+ * Copyright (c) 2018 Stichting Yona Foundation This Source Code Form is subject to the terms of the Mozilla Public License, v.
+ * 2.0. If a copy of the MPL was not distributed with this file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *******************************************************************************/
+package nu.yona.server.util;
+
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import nu.yona.server.exceptions.YonaException;
+
+public class Require
```

Comment:
As discussed, I'm actually quite happy with this as it reduces quite some lines of code.

What I forgot to mention: phrasing it positively also increases the readability for me (instead of ``if (!a) then throw``, write ``Require.that(a, ...)``.